### PR TITLE
Subgroup visibility

### PR DESCRIPTION
**Pull request by jczacharia**
This PR adds the feature: the ability to hide/show specific subgroups.

Demo: http://plnkr.co/edit/qJSULoHlFF7GvcoWhhsK

### CODE REVIEW

**Comment by yotamberk** [comment]
Hi @jczacharia , I'm getting weird behavior from your example. It seems to hide the subgroup correctly but a redraw is not called, so the stacking and height calculations are not applied.
Are you seeing this too?

**Comment by jczacharia** [comment]
Yes, I did notice that. I am not sure that that is because of my changes. It seems like that was a present bug.

File: examples/timeline/groups/subgroupsVisibility.html [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          id: 'background_all',
+          start: '2014-01-31',
+          end: '2014-02-02',
+          type: 'background',
+          className: 'positive'
+        }
+      ]);
+
+      var container = document.getElementById('visualization');
+      var stackingStatus = document.getElementById('stackingStatus');
```
Comment (yotamberk) [inline examples/timeline/groups/subgroupsVisibility.html]: Remove. Not used.

File: examples/timeline/groups/subgroupsVisibility.html [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      var stackSubgroupsStatus = document.getElementById('stackSubgroupsStatus');
+      var options = {
+        // orientation:'top'
+        start: '2014-01-10',
+        end: '2014-02-10',
+        editable: true,
+        stack: false,
+        stackSubgroups: true,
+        selectable: true
```
Comment (yotamberk) [inline examples/timeline/groups/subgroupsVisibility.html]: Remove. Irrelevant to this example.

File: examples/timeline/groups/subgroupsVisibility.html [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+      // containing an ISO date. The fields will be outputted as ISO dates
+      // automatically getting data from the DataSet via items.get().
+      var items = new vis.DataSet({
+        type: { start: 'ISODate', end: 'ISODate' }
+      });
+      var groups = new vis.DataSet([
+        {
+          id: 'bar',
+          content: 'bar',
+          subgroupOrder: function(a, b) {
```
Comment (yotamberk) [inline examples/timeline/groups/subgroupsVisibility.html]: Remove. Irrelevant to this example.

File: examples/timeline/groups/subgroupsVisibility.html [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+            return a.subgroupOrder - b.subgroupOrder;
+          },
+          subgroupStack: { sg_1: true, sg_2: true, sg_3: true },
```
Comment (yotamberk) [inline examples/timeline/groups/subgroupsVisibility.html]: Remove. Irrelevant to this example.

File: examples/timeline/groups/subgroupsVisibility.html [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+          subgroupVisibility: { sg_1: true, sg_2: true, sg_3: true }
+        },
+        {
+          id: 'foo',
+          content: 'foo',
+          subgroupOrder: 'subgroupOrder' // this group has no subgroups but this would be the other method to do the sorting.
```
Comment (yotamberk) [inline examples/timeline/groups/subgroupsVisibility.html]: Remove. Irrelevant to this example.

File: lib/timeline/component/Group.js [hunk 6]
```diff
@@ -528,7 +540,7 @@ class Group {
       util.forEach(this.visibleItems, item => {
         if (item.data.subgroup !== undefined) {
           me.subgroups[item.data.subgroup].height = Math.max(me.subgroups[item.data.subgroup].height, item.height + margin.item.vertical);
-          me.subgroups[item.data.subgroup].visible = true;
+          me.subgroups[item.data.subgroup].visible = this.subgroupVisibility[item.data.subgroup] !== undefined ? this.subgroupVisibility[item.data.subgroup] : true;
```
Comment (yotamberk) [inline lib/timeline/component/Group.js]: See https://github.com/visjs/vis-timeline/pull/69#discussion_r319733787

**Comment by mojoaxel** [comment]
>  think we need to clean the example as much as possible to show only the feature added.

I think the example is ok now!

**Pull request merged**